From 096924fc94d16841a1e60b385962546af92df6ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 5 Oct 2026 04:31:01 +0200
Subject: [PATCH] system-info: Make the window scrollable

This allows the window to fit into smaller screens. Without this, the
bottom of the window is off screen on a standard 1024x768 display.
Also, allow the user to resize the window.
---
 capplets/system-info/mate-system-info.c  |   1 -
 capplets/system-info/mate-system-info.ui | 283 ++++++++++++-----------
 2 files changed, 153 insertions(+), 131 deletions(-)

diff --git a/capplets/system-info/mate-system-info.c b/capplets/system-info/mate-system-info.c
index 00b0d8a37..6a23124c8 100644
--- a/capplets/system-info/mate-system-info.c
+++ b/capplets/system-info/mate-system-info.c
@@ -769,7 +769,6 @@ mate_system_info_new (void)
                            NULL);
     gtk_window_set_title (GTK_WINDOW (dialog), _("Mate System Info"));
     gtk_widget_set_size_request (GTK_WIDGET (dialog), 600, 500);
-    gtk_window_set_resizable  (GTK_WINDOW (dialog), FALSE);
 
     return dialog;
 }
diff --git a/capplets/system-info/mate-system-info.ui b/capplets/system-info/mate-system-info.ui
index d4cafa76a..84c1383db 100644
--- a/capplets/system-info/mate-system-info.ui
+++ b/capplets/system-info/mate-system-info.ui
@@ -6,151 +6,174 @@
     <property name="icon-name">mate-desktop</property>
     <property name="type_hint">dialog</property>
 
-      <child internal-child="vbox">
-        <object class="GtkBox">
+    <child internal-child="vbox">
+      <object class="GtkBox">
         <property name="visible">True</property>
         <property name="can_focus">False</property>
-        <property name="valign">center</property>
-        <property name="spacing">30</property>
         <property name="orientation">vertical</property>
-        <child>
-          <object class="GtkImage" id="logo_image">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-          </object>
-          <packing>
-            <property name="expand">False</property>
-            <property name="fill">False</property>
-          </packing>
-        </child>
 
         <child>
-          <object class="GtkListBox">
+          <object class="GtkScrolledWindow" id="scrolled_window">
             <property name="visible">True</property>
-            <property name="selection-mode">none</property>
-            <style>
-              <class name="frame" />
-            </style>
-
-            <!-- Device name -->
-            <child>
-              <object class="GtkListBoxRow" id="hostname_row">
-                <property name="visible">True</property>
-              </object>
-            </child>
-
-          </object>
-        </child>
-
-        <child>
-          <object class="GtkListBox" id="hardware_box">
-            <property name="visible">True</property>
-            <property name="selection-mode">none</property>
-            <style>
-              <class name="frame" />
-            </style>
-
-            <!-- Hardware Model -->
-            <child>
-              <object class="GtkListBoxRow" id="hardware_model_row">
-                <property name="visible">False</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-
-            <!-- Memory -->
-            <child>
-              <object class="GtkListBoxRow" id="memory_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-
-            <!-- Processor -->
-            <child>
-              <object class="GtkListBoxRow" id="processor_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
+            <property name="can_focus">False</property>
+            <property name="hexpand">True</property>
+            <property name="vexpand">True</property>
+            <property name="hscrollbar-policy">never</property>
+            <property name="vscrollbar-policy">automatic</property>
 
-            <!-- Graphics -->
             <child>
-              <object class="GtkListBoxRow" id="graphics_row">
+              <object class="GtkBox">
                 <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
+                <property name="can_focus">False</property>
+                <property name="valign">start</property>
+                <property name="spacing">30</property>
+                <property name="orientation">vertical</property>
+
+                <child>
+                  <object class="GtkImage" id="logo_image">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">False</property>
+                  </packing>
+                </child>
+
+                <child>
+                  <object class="GtkListBox">
+                    <property name="visible">True</property>
+                    <property name="selection-mode">none</property>
+                    <style>
+                      <class name="frame" />
+                    </style>
+
+                    <!-- Device name -->
+                    <child>
+                      <object class="GtkListBoxRow" id="hostname_row">
+                        <property name="visible">True</property>
+                      </object>
+                    </child>
+
+                  </object>
+                </child>
+
+                <child>
+                  <object class="GtkListBox" id="hardware_box">
+                    <property name="visible">True</property>
+                    <property name="selection-mode">none</property>
+                    <style>
+                      <class name="frame" />
+                    </style>
+
+                    <!-- Hardware Model -->
+                    <child>
+                      <object class="GtkListBoxRow" id="hardware_model_row">
+                        <property name="visible">False</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Memory -->
+                    <child>
+                      <object class="GtkListBoxRow" id="memory_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Processor -->
+                    <child>
+                      <object class="GtkListBoxRow" id="processor_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Graphics -->
+                    <child>
+                      <object class="GtkListBoxRow" id="graphics_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Disk Capacity -->
+                    <child>
+                      <object class="GtkListBoxRow" id="disk_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                  </object>
+                </child>
+
+                <child>
+                  <object class="GtkListBox" id="os_box">
+                    <property name="visible">True</property>
+                    <property name="selection-mode">none</property>
+                    <style>
+                      <class name="frame" />
+                    </style>
+
+                    <!-- OS Name -->
+                    <child>
+                      <object class="GtkListBoxRow" id="os_name_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- OS Type -->
+                    <child>
+                      <object class="GtkListBoxRow" id="os_type_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- MATE Version -->
+                    <child>
+                      <object class="GtkListBoxRow" id="mate_version_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Windowing System -->
+                    <child>
+                      <object class="GtkListBoxRow" id="windowing_system_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Virtualization -->
+                    <child>
+                      <object class="GtkListBoxRow" id="virtualization_row">
+                        <property name="visible">False</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                    <!-- Kernel -->
+                    <child>
+                      <object class="GtkListBoxRow" id="kernel_row">
+                        <property name="visible">True</property>
+                        <property name="activatable">False</property>
+                      </object>
+                    </child>
+
+                  </object>
+                </child>
 
-            <!-- Disk Capacity -->
-            <child>
-              <object class="GtkListBoxRow" id="disk_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
               </object>
             </child>
-
           </object>
         </child>
 
-        <child>
-          <object class="GtkListBox" id="os_box">
-            <property name="visible">True</property>
-            <property name="selection-mode">none</property>
-            <style>
-              <class name="frame" />
-            </style>
-
-            <!-- OS Name -->
-            <child>
-              <object class="GtkListBoxRow" id="os_name_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-
-            <!-- OS Type -->
-            <child>
-              <object class="GtkListBoxRow" id="os_type_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-
-            <!-- MATE Version -->
-            <child>
-              <object class="GtkListBoxRow" id="mate_version_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-
-            <!-- Windowing System -->
-            <child>
-              <object class="GtkListBoxRow" id="windowing_system_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-
-            <!-- Virtualization -->
-            <child>
-              <object class="GtkListBoxRow" id="virtualization_row">
-                <property name="visible">False</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-            
-            <!-- Kernel -->
-            <child>
-              <object class="GtkListBoxRow" id="kernel_row">
-                <property name="visible">True</property>
-                <property name="activatable">False</property>
-              </object>
-            </child>
-          </object>
-        </child>
-     </object>
+      </object>
     </child>
   </template>
 </interface>
